classes/form/CustomerAddressForm.php

```diff
@@ -217,4 +209,73 @@ function (FormField $item) {
             'formFields' => $formFields,
         ];
     }
+
+    /**
+     * Performs validation on field values.
+     * Returns true if all field values are correct, false otherwise.
+     *
+     * @return bool
+     */
+    private function validateFieldsValues(): bool
+    {
+        $isValid = true;
+
+        $isValid &= $this->validatePostcode();
+        $isValid &= $this->validateField('firstname', 'isName', $this->translator->trans(
+            'Invalid name',
+            [],
+            'Shop.Forms.Errors'
+        ));
+        $isValid &= $this->validateField('lastname', 'isName', $this->translator->trans(
+            'Invalid name',
+            [],
+            'Shop.Forms.Errors'
+        ));
+        $isValid &= $this->validateField('city', 'isCityName', $this->translator->trans(
+            'Invalid format.',
+            [],
+            'Shop.Forms.Errors'
+        ));
+
+        return (bool) $isValid;
+    }
+
+    private function validatePostcode(): bool
+    {
+        if (($postcode = $this->getField('postcode'))) {
+            if ($postcode->isRequired()) {
+                $country = $this->formatter->getCountry();
+                if (!$country->checkZipCode($postcode->getValue())) {
+                    $postcode->addError($this->translator->trans(
+                        'Invalid postcode - should look like "%zipcode%"',
+                        ['%zipcode%' => $country->zip_code_format],
+                        'Shop.Forms.Errors'
+                    ));
+
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    private function validateField(string $fieldName, string $validationFunction, string $validationFailMessage): bool
```

Comment:
PHPDoc might be useful for developers using an IDE that does not support type hinting